libc/src/stdlib/qsort_r.cpp

```diff
@@ -22,8 +22,14 @@ LLVM_LIBC_FUNCTION(void, qsort_r,
   if (array == nullptr || array_size == 0 || elem_size == 0)
     return;
   internal::Comparator c(compare, arg);
-  internal::quicksort(internal::Array(reinterpret_cast<uint8_t *>(array),
-                                      array_size, elem_size, c));
+  auto arr = internal::Array(reinterpret_cast<uint8_t *>(array), array_size,
+                             elem_size, c);
+
+#if LIBC_QSORT_IMPL == LIBC_QSORT_QUICK_SORT
```

Comment:
would it make sense to have this switch be in `qsort_util` and have these functions call something like `internal::sort`?

Reply:
I refactor `qsort_util.h` and move the selection logic into `qsort_util.h`.